cpp/CMakeLists.txt

```diff
@@ -299,6 +299,7 @@ add_library(
   src/hash/hashing.cu
   src/hash/md5_hash.cu
   src/hash/murmur_hash.cu
+  src/hash/spark_murmur_hash.cu
```

Comment:
Should this be named int32_murmur_hash?

For consistency it would be great to have the files named based on the algorithm, and not the consumer

Reply:
> For consistency it would be great to have the files named based on the algorithm, and not the consumer.

I agree completely. I would say this is an unfortunate exception. This algorithm is specifically designed to match a Spark implementation and there’s no concise or generic way (that avoids the word “Spark”) to describe the ways in which it differs from the standard MurmurHash3_32 implementation. There is specialized tail processing in the hash functor, specializations for various decimal types, special behavior for serially using the output hash as an input seed for the next item, and ignorance of null values among other differences. The int32 vs uint32 return type issue is a problem which I plan to revisit in later work, and is not the essential difference here.

Reply:
We should provide more details in the `device_spark_row_hasher` documentation on how it differs compared to the standard `MurmurHash3_32`